From efd27be644cb0679377690e41469c57d31c4e670 Mon Sep 17 00:00:00 2001
From: Ben Walton <bwalton@opencsw.org>
Date: Wed, 13 Jun 2012 04:12:40 +0200
Subject: [PATCH] Ensure we have FIONBIO defined

Include sys/filio.h in files that use the FIONBIO constant.

Signed-off-by: Ben Walton <bwalton@opencsw.org>
---
 lib5250/sslstream.c |    4 ++++
 lib5250/telnetstr.c |    4 ++++
 lib5250/utility.c   |    4 ++++
 3 files changed, 12 insertions(+)

diff --git a/lib5250/sslstream.c b/lib5250/sslstream.c
index 7181566..d4d0342 100644
--- a/lib5250/sslstream.c
+++ b/lib5250/sslstream.c
@@ -30,6 +30,10 @@
 #include <openssl/err.h>
 #include <time.h>
 
+#if defined (__SVR4) && defined (__sun)
+#include <sys/filio.h>
+#endif
+
 static int ssl_stream_get_next(Tn5250Stream *This,unsigned char *buf,int size);
 static void ssl_stream_do_verb(Tn5250Stream * This, unsigned char verb, unsigned char what);
 static int ssl_stream_host_verb(Tn5250Stream * This, unsigned char verb,
diff --git a/lib5250/telnetstr.c b/lib5250/telnetstr.c
index 9ad2624..7fa2688 100644
--- a/lib5250/telnetstr.c
+++ b/lib5250/telnetstr.c
@@ -21,6 +21,10 @@
  */
 #include "tn5250-private.h"
 
+#if defined (__SVR4) && defined (__sun)
+#include <sys/filio.h>
+#endif
+
 static int telnet_stream_get_next(Tn5250Stream * This, unsigned char *buf, int size);
 static void telnet_stream_do_verb(Tn5250Stream * This, unsigned char verb, unsigned char what);
 static int telnet_stream_host_verb(Tn5250Stream * This, unsigned char verb,
diff --git a/lib5250/utility.c b/lib5250/utility.c
index 9f55b36..c736697 100644
--- a/lib5250/utility.c
+++ b/lib5250/utility.c
@@ -22,6 +22,10 @@
 #include "tn5250-private.h"
 #include "transmaps.h"
 
+#if defined (__SVR4) && defined (__sun)
+#include <sys/filio.h>
+#endif
+
 static char mapfix[256];
 static char mapfix2[256];
 static char mapfix3[256];
-- 
1.7.10.3

